packages/jest-mock/src/__tests__/index.test.ts

```diff
@@ -707,6 +707,15 @@ describe('moduleMocker', () => {
       ]);
     });
 
+    it('supports mocking for specific args', () => {
+      const fn = moduleMocker.fn();
+      fn.mockReturnValue('generic output');
+      fn.mockWhen('special input').mockReturnValue('special output');
+
+      expect(fn('arbitrary')).toBe('generic output');
+      expect(fn('special input')).toBe('special output');
```

Comment:
Maybe add some more calls to this? to be sure it's consistent?

```suggestion
      expect(fn('special input')).toBe('special output');
      expect(fn('arbitrary')).toBe('generic output');
      expect(fn('arbitrary')).toBe('generic output');
      expect(fn('special input')).toBe('special output');
```